hazelcast-spring/src/main/java/com/hazelcast/spring/cache/HazelcastCacheManager.java

```diff
@@ -70,11 +83,59 @@ public Cache getCache(String name) {
         return cacheNames;
     }
 
-    public void setHazelcastInstance(final HazelcastInstance hazelcastInstance) {
-        this.hazelcastInstance = hazelcastInstance;
+    private long calculateCacheReadTimeout(String name) {
+        long timeout = 0;
+        if (getReadTimeoutMap().containsKey(name)) {
+            Long v = getReadTimeoutMap().get(name);
+            if (v != null) {
+                timeout = v;
+            }
+        } else {
+            timeout = defaultReadTimeout;
+        }
+        return timeout;
     }
 
     public HazelcastInstance getHazelcastInstance() {
         return hazelcastInstance;
     }
+
+    public void setHazelcastInstance(final HazelcastInstance hazelcastInstance) {
+        this.hazelcastInstance = hazelcastInstance;
+    }
+
+    /**
+     * Return default cache value retrieval timeout in milliseconds.
+     */
+    public long getDefaultReadTimeout() {
+        return defaultReadTimeout;
+    }
+
+    /**
+     * Set default cache value retrieval timeout. Applies to all caches, if not defined a cache specific timeout.
+     * @param defaultReadTimeout default cache retrieval timeout in milliseconds. 0 or negative values disable timeout.
+     */
+    public void setDefaultReadTimeout(long defaultReadTimeout) {
+        this.defaultReadTimeout = defaultReadTimeout;
+    }
+
+    /**
+     * Return cache-specific value retrieval timeouts. Map keys are cache names, values are cache retrieval timeouts in
+     * milliseconds.
+     */
+    public Map<String, Long> getReadTimeoutMap() {
+        if (readTimeoutMap == null) {
+            readTimeoutMap = Collections.emptyMap();
+        }
```

Comment:
This can be initialized in `field` definition or constructor unless we really want to do it on-demand.